apps/prairielearn/src/tests/fileEditor.test.ts

```diff
@@ -1066,15 +1071,48 @@ function doFiles(data) {
 
         testDeleteFile({
           url: data.url + '/' + encodePath(path.join(data.path, data.testFilesDir, 'subdir')),
-          id: 0,
           path: path.join(data.path, data.testFilesDir, 'subdir', 'testfile.txt'),
         });
       });
     }
+    describe('Files with % in name', function () {
+      testUploadFile({
+        adminUrl: data.url,
+        url: data.url,
+        path: path.join(data.path, 'test%file.txt'),
+        id: 'New',
+        contents: 'This is a line of text in a file with percent.',
+        filename: 'test%file.txt',
+      });
+
+      // TODO Rename currently has very restrictive naming conventions that
+      // don't allow for this kind of name. Once this is removed it should be
+      // possible to enable the test below.
+
+      // testRenameFile({
+      //   url: data.url,
+      //   id: data.index,
+      //   path: path.join(data.path, 'sub%dir', 'test%file.txt'),
+      //   contents: 'This is a line of text in a file with percent.',
+      //   new_file_name: path.join('sub%dir', 'test%file.txt'),
+      // });
+
+      testDeleteFile({
+        url: data.url + '/' + encodePath(data.path),
+        path: path.join(data.path, 'test%file.txt'),
+      });
+    });
   });
 }
 
-function testUploadFile(params) {
+function testUploadFile(params: {
+  adminUrl: string;
```

Comment:
Assuming I got the purpose of this right, maybe call this `fileViewBaseUrl` or something like that?

Reply:
This function is used to upload a new file. I just added the check that, after uploading, the file can be viewed and downloaded with the same contents. I think that "upload" is the main feature, the view/download is a secondary goal, so I kept the name.

Reply:
I'm not referring to the function name, which is fine; I'm talking about `adminUrl` (the line I commented on).

Reply:
Sorry, misunderstood. Renamed in f57fe2b69.